src/extension-test/unit/cursor-doc/token-cursor-test.ts

```diff
@@ -28,6 +28,20 @@ describe('Token Cursor', () => {
       cursor.forwardSexp();
       expect(cursor.offsetStart).toBe(b.selection.anchor);
     });
+    it('forwardSexp with newline', () => {
+      const a = docFromTextNotation('|(a\n(b))');
+      const b = docFromTextNotation('(a\n(b))|');
+      const cursor: LispTokenCursor = a.getTokenCursor(a.selection.anchor);
+      cursor.forwardSexp();
+      expect(cursor.offsetStart).toBe(b.selection.anchor);
+    });
+    it('forwardSexp with newline (MS-Windows)', () => {
+      const a = docFromTextNotation('|(a\r\n(b))');
+      const b = docFromTextNotation('(a\r\n(b))|');
+      const cursor: LispTokenCursor = a.getTokenCursor(a.selection.anchor);
+      cursor.forwardSexp();
+      expect(cursor.offsetStart).toBe(b.selection.anchor);
+    });
```

Comment:
Thanks! Great to have this covered.

Reply:
Thanks, this is the first thing I tried to reproduce as a test.

Reply:
We have an improved docFromTextNotation utility and some more goodies around it in a PR branch that has gone a bit stale. We should salvage that part of it.